### PR TITLE
Remove deploy step for macOS CI

### PR DESCRIPTION
**Pull request by vespakoen**
Removing deploy step from macOS CI `test` stage.
Will also add the `if: tag IS present` condition if the first build succeeds.

### CODE REVIEW

**Comment by ppd** [comment]
Suggestion: commit message `Travis: remove deploy step from macOS CI test stage`

**Comment by vespakoen** [comment]
Allright! I will squash those commits before merge.

- I also added OpenMP
- Forced it to build with `BUILD_TYPE=Release` to try that out
- Updated the xcode version in the travis build

Let's see what happens, i'll use this branch to try out some things and hopefully land on a nice setup for macOS soon.

File: src/drawentity.cpp [hunk 1]
```diff
@@ -315,9 +315,13 @@ void Entity::ComputeInterpolatingSpline(SBezierList *sbl, bool periodic) const {
             } else {
                 // The wrapping would work, except when n = 1 and everything
                 // wraps to zero...
-                if(i > 0)     bm.A[i][i - 1] = eq.x;
-                /**/          bm.A[i][i]     = eq.y;
-                if(i < (n-1)) bm.A[i][i + 1] = eq.z;
+                if(i > 0) {
```
Comment (ruevs) [inline src/drawentity.cpp:318]: @vespakoen @phkahler 
These changes (I am not sure if they make sense) do not belong in this commit!

File: src/solvespace.h [hunk 2]
```diff
@@ -513,7 +513,7 @@ class SolveSpaceUI {
     GraphicsWindow              GW;
 
     // The state for undo/redo
-    typedef struct {
+    typedef struct UndoState {
```
Comment (ruevs) [inline src/solvespace.h:516]: Yes, I've been meaning to make a pull request for this annoying warning.
But, again, in my opinion it does not belong in this his commit.

File: src/mouse.cpp [hunk 3]
```diff
@@ -624,24 +624,24 @@ void GraphicsWindow::MouseRightUp(double x, double y) {
         }
         if(gs.withEndpoints > 0) {
             menu->AddItem(_("Select Edge Chain"),
-                          [this]() { MenuEdit(Command::SELECT_CHAIN); });
+                []() { MenuEdit(Command::SELECT_CHAIN); });
```
Comment (ruevs) [inline src/mouse.cpp:627]: Same here. This whole file...

**Review by ruevs — COMMENTED** [review]
Please be more careful what you include in the pull requests.

File: .travis/build-macos.sh [hunk 4]
```diff
@@ -1,9 +1,23 @@
...
+if echo $TRAVIS_TAG | grep ^v; then BUILD_TYPE=Release; else BUILD_TYPE=Debug; fi
 
-mkdir build
+mkdir build || true
 cd build
-cmake -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9 -DCMAKE_BUILD_TYPE=$BUILD_TYPE ..
-make -j2
-make test_solvespace
+
+export BUILD_TYPE=Release
```
[the remark this answers is not in the document]
Comment (vespakoen) [reply .travis/build-macos.sh]: This was just for checking if the build would work in Release mode, I changed it back to the default (automatic switching between `Debug` and `RelWithDebInfo` based on the presence of `$TRAVIS_TAG`

**Comment by vespakoen** [comment]
@ruevs Points taken, sorry, I will split these up better in the future.

There were a lot of warnings, and I got some LLVM (v11) crashes / parse errors, so I thought maybe removing those warnings would get me further.

File: .travis/build-macos.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (ruevs) [reply .travis/build-macos.sh]: I noticed too late. Sorry.

**Comment by phkahler** [comment]
Wondering if we should unconditionally build with OpenMP now. Maybe the option could disable it - that's sometimes helpful in development. We were not using it before because whitequark didn't want users to need the extra DLL on Windows where we ship with no installer (just an exe). Jonathan is OK with requiring that, as am I. So if we're going to enable it for releases, should we just build with it unless set otherwise?

LTO is different because it takes a lot (3 to 4x)  longer to build and offers only 30% faster code (in the one test I measured). So I still think that has to be for release builds only.
